app/src/main/java/app/nexd/android/ui/helper/transcript/TranscriptViewModel.kt

```diff
@@ -84,45 +85,29 @@ class TranscriptViewModel(private val api: Api) : ViewModel() {
         compositeDisposable.add(disposable)
     }
 
-    fun reloadCalls() {
-        val calls = api.phoneControllerGetCalls(
+    @SuppressLint("CheckResult")
+    fun transcriptCall() {
+        resetData()
+
+        api.phoneControllerGetCalls(
             null,
-            10,
+            1,
             false,
             null,
             null,
             null
         )
-
-        val disposable = calls
+            .firstOrError()
```

Comment:
That's correct. But transforming the Observable to a single will automatically terminate, so I don't have to add it to an composite disposable.

Reply:
This is not exact. 

1. Both would terminate in same manner because our Observable comes from a rest call and thus can emit only one event. Then it will complete on itself.
2. You always have to use CompositeDisposable because it is not about completion in general. Disposable allows you to cancel the stream before it completes on itself. In context of UI this is crucial if the subscription will try to change something in the ui (e.g. display a text). If the rest requests takes some time because of slow backend or bad internet connection, this change could happen after user has switched the screen or even minimized the app. Changes on unattached fragments/views will cause an error and lead to a crash.
3. By subscribing to a stream in e.g. a ViewModel you register a reference of the Observer/Consumer in the stream. This will lead to memory leaks of the ViewModel if not disposed correctly. You switch the screen but the ViewModel can't be deleted by Garbage Collector because there is still a reference to it from some streams.

Because of that you mostly want to handle your disposables (e.g. disposing them in onCleare). The only exception is if you really want this stream to continue even if the screen is already gone. In this case you have to pay attention which lifecycle you want to use for the subscription.